src/chanjo2/endpoints/samples.py

```diff
@@ -1,45 +1,81 @@
 from pathlib import Path
 from typing import List
 
+from chanjo2.constants import BAD_REQUEST, NOT_FOUND
+from chanjo2.crud import samples as crud_samples
 from chanjo2.dbutil import get_session
-from chanjo2.meta.handle_bed import parse_bed
-from chanjo2.models.pydantic_models import CoverageInterval, SampleCreate, SampleRead
-from fastapi import APIRouter, Depends, File, HTTPException, Query, status
-from sqlmodel import Session, select
+from chanjo2.models import pydantic_models, sql_models
+from fastapi import APIRouter, Depends, HTTPException, status
+from sqlmodel import Session
 
 router = APIRouter()
 
 
-@router.post("/sample_create/", response_model=SampleRead)
-def create_sample(*, session: Session = Depends(get_session), sample: SampleCreate):
+### Case endpoints
+@router.post("/cases/", response_model=pydantic_models.CaseRead)
+def create_case(case: pydantic_models.CaseCreate, db: Session = Depends(get_session)):
+    """Endpoint used to add a case to the database"""
+    db_case = crud_samples.get_case(db, case_name=case.name)
+    if db_case:
+        raise HTTPException(status_code=BAD_REQUEST, detail="Case already registered")
+    return crud_samples.create_case(db=db, case=case)
+
+
+@router.get("/cases/", response_model=List[pydantic_models.CaseRead])
+def read_cases(skip: int = 0, limit: int = 100, db: Session = Depends(get_session)):
+    """Endpoint used to fetch all existing cases from the database"""
+    cases = crud_samples.get_cases(db, skip=skip, limit=limit)
+    return cases
+
+
+@router.get("/cases/{case_name}", response_model=pydantic_models.CaseRead)
+def read_case(case_name: str, db: Session = Depends(get_session)):
+    """Endpoint used to fetch one cases from the database by providing its ID"""
+    db_case = crud_samples.get_case(db, case_name=case_name)
+    if db_case is None:
+        raise HTTPException(status_code=NOT_FOUND, detail="Case not found")
+    return db_case
+
+
+### Sample endpoints
+@router.post("/samples/", response_model=pydantic_models.SampleRead)
+def create_sample_for_case(
+    sample: pydantic_models.SampleCreate,
+    db: Session = Depends(get_session),
+):
+    """Endpoint used to add a case sample to the database"""
```

Comment:
```suggestion
    """Add a sample to a case in the database."""
```
If the route is in plural why is there no loop too add multiple samples to a case? Am I missing something?

Reply:
I see your point. It's intuitive to use `sampleS` for stuff that do things with many samples and `sample` for stuff that returns or creates just one sample. 
I did this on purpose actually. since I've read that for consistency they recommend to use nouns in their plural form to represent resource. [Here](https://auth0.com/blog/best-practices-for-flask-api-development/) for instance, or [here](https://restfulapi.net/resource-naming/). Even on the FASTApi tutorial they use the plural naming in the URL, even if you are just returning one object: 

<img width="965" alt="image" src="https://user-images.githubusercontent.com/28093618/216975825-94709f7e-fde4-4d21-8b0c-24f4e2516855.png">

I'd like to keep it like this if it's OK with you. It's just a convention eh